typhon/files/handlers/common.py

```diff
@@ -715,7 +716,10 @@ def _load_group(ds, path, group, fields):
             for var_name, var in group.variables.items():
                 if fields is None or path + var_name in fields:
                     dims = [dim_map[dim] for dim in var.dimensions]
-                    ds[path + var_name] = dims, var[:], dict(var.__dict__)
+                    if len(dims) == 0 and var[:] is np.ma.masked:
```

Comment:
I would have used
```python
if len(dims) == 0 and isinstanceof(var[:], np.ma.MaskedArray):
```
 but it is very possible that both are equivalent for all possible cases.

Reply:
Ah, I didn't realise that `ma.masked` is a `MaskedArray`.  I assumed it was a [numpy scalar](https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.scalars.html), for it's not actually an array.  I'm not sure that the two are equivalent, and I'm not sure exactly what combination of circumstances causes the problem.

Reply:
I guess that every ``np.ma.masked`` is an ``np.ma.MaskedArray`` but I am not sure if my condition could be ``True`` for other, unintended, cases. I am fine with your approach, though 😉

Reply:
`isinstance(ma.MaskedArray([0]), ma.MaskedArray)` would be `True` (was your `isinstanceof` a typo or does it refer to a function I don't know?), I don't know if there can be a scalar `MaskedArray` and if there is, I don't know if it would trigger this bug!

Reply:
It was a "typo" (it sounded right when typing the comment 😄). It looks like both conditions solve the problem for now so let's just choose one of them. You could leave a comment that states that we are not sure for our future selfs 😉

Reply:
Correction — `ma.masked` is not a scalar, but a 0-dimensional array...

Reply:
`ma.MaskedArray(data=0.0, mask=True)` is also a scalar instance of `ma.MaskedArray`, but `is not ma.masked`.
However, `ds['a'] = (), ma.MaskedArray(data=0.0, mask=True), {}` actually works (and sets it to nan).  So my check does seem to catch the problematic case, but I'm not sure if there is supposed to be a difference between `ma.masked` and `ma.MaskedArray(data=0.0, mask=True)`, as both are 0-dimensional arrays with the same underlying data and the same type; `ma.masked.astype(ma.masked.dtype)` in fact gives `ma.MaskedArray(data=0.0, mask=True)`...

Reply:
From [the `numpy` documentation](https://docs.scipy.org/doc/numpy-1.15.1/reference/maskedarray.baseclass.html#numpy.ma.masked):

> The masked constant is a special case of MaskedArray, with a float datatype and a null shape. It is used to test whether a specific entry of a masked array is masked, or to mask one or several entries of a masked array.

And testing is what I'm doing here, so I think mine is the correct approach.  Of course, one could argue that `ds['a'] = (), ma.masked, {}` should be valid, but that would be a question for xarray.